src/Gold.Redis/Gold.Redis.HighLevelClient/Db/RedisCommandsExecutor.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Gold.Redis.Common.Interfaces.Communication;
+using Gold.Redis.Common.Interfaces.Db;
+using Gold.Redis.Common.Models.Commands;
+using Newtonsoft.Json;
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Gold.Redis.HighLevelClient.Db
+{
+    public class RedisCommandsExecutor : IRedisCommandExecutor
+    {
+        private readonly IRedisCommandHandler _redisCommandHandler;
+        private readonly JsonSerializer _jsonSerializer;
+
+        public RedisCommandsExecutor(
+            IRedisCommandHandler redisCommandHandler,
+            JsonSerializerSettings serializerSettings = null)
+        {
+            _redisCommandHandler = redisCommandHandler;
+            _jsonSerializer = JsonSerializer.Create(serializerSettings);
+        }
+        public async Task<T> Execute<T>(Command command)
+        {

```

Comment:
we should think about moving the JSON part to something more generic. 
Each client will have different usage of the library so we dont want to limit them